src/utils/instructionDecoder.ts

```diff
@@ -319,81 +322,115 @@ export const decodeInstruction = (instruction: TransactionInstruction): DecodedD
 
-  // TODO: Need to Decode for Token Account Creation and Transfer Instruction which bundle in 1 Transaction.
-  let intrestedInstructionidx = 0;
-  const instructionLength = instructions.length;
+    // TODO: Need to Decode for Token Account Creation and Transfer Instruction which bundle in 1 Transaction.
+    let interestedTransferInstructionidx = 0;
+    let interestedBurnInstructionidx = -1;
+    const instructionLength = instructions.length;
 
-  if (instructionLength > 1 && instructionLength <= 3) {
-    const createInstructionIdx = instructions.findIndex((inst) => {
-      if (inst.programId.equals(ASSOCIATED_TOKEN_PROGRAM_ID)) {
-        return inst.data.length === 0;
-      }
-      return false;
-    });
-    if (createInstructionIdx >= 0) {
-      const transferIdx = instructions.findIndex((inst) => {
-        if (inst.programId.equals(TOKEN_PROGRAM_ID)) {
-          const parseInst = decodeTokenInstruction(inst);
-          return ["transfer", "transferChecked"].includes(parseInst.type);
+    if (instructionLength > 1 && instructionLength <= 3) {
+      const createInstructionIdx = instructions.findIndex((inst) => {
+        if (inst.programId.equals(ASSOCIATED_TOKEN_PROGRAM_ID)) {
+          return inst.data.length === 0;
         }
         return false;
       });
-      intrestedInstructionidx = transferIdx;
+      if (createInstructionIdx >= 0) {
+        const transferIdx = instructions.findIndex((inst) => {
+          if (inst.programId.equals(TOKEN_PROGRAM_ID)) {
+            const parseInst = decodeTokenInstruction(inst);
+            return ["transfer", "transferChecked"].includes(parseInst.type);
+          }
+          return false;
+        });
+        interestedTransferInstructionidx = transferIdx;
+      } else {
+        const burnIndex = instructions.findIndex((inst) => {
+          if (inst.programId.equals(TOKEN_PROGRAM_ID)) {
+            const parseInst = decodeTokenInstruction(inst);
+            return ["burn", "burnChecked"].includes(parseInst.type);
+          }
+          return false;
+        });
+        interestedBurnInstructionidx = burnIndex;
+      }
     }
-  }
 
-  // Expect SPL token transfer transaction have only 1 instruction
-  if (instructions.length === 1 || intrestedInstructionidx > 0) {
-    if (TOKEN_PROGRAM_ID.equals(instructions[intrestedInstructionidx].programId)) {
-      const decoded = decodeTokenInstruction(instructions[intrestedInstructionidx]);
-      // There are transfer and transferChecked type
-      if (decoded.type.includes("transfer")) {
-        const from = new PublicKey(decoded.data.source || "").toBase58();
-        const to = new PublicKey(decoded.data.destination || "").toBase58();
+    // Expect SPL token transfer transaction have only 1 instruction
+    if (instructions.length === 1 || interestedTransferInstructionidx > 0 || interestedBurnInstructionidx > -1) {
+      if (TOKEN_PROGRAM_ID.equals(instructions[interestedTransferInstructionidx].programId)) {
```

Comment:
Shouldn't the burn instruction index is used to checked ?
another if check should be created